pymc/func_utils.py

```diff
@@ -0,0 +1,142 @@
+#   Copyright 2021 The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+import warnings
+
+from typing import Dict, Optional
+
+import aesara.tensor as aet
+import numpy as np
+
+from scipy import optimize
+
+import pymc as pm
+
+__all__ = ["find_optim_prior"]
+
+
+def find_optim_prior(
```

Comment:
I think functionality wise this is great, but I wonder about the name. 

Why optimal? What about something like `find_prior_in_range` or `find_constrained_prior` or better yet something else.

Optimal has a connotation that such priors are special, but it seems the `optim` comes from the mere fact that we are using an optimization procedure to find them.

Reply:
Actually the "optim" stands for "optimized", so I should rename the function to `find_optimized_prior` if that sounds good to you.
Otherwise, I also like `find_constrained_prior`. `better_yet_something_else` does have a nice ring to it, but TBH is way too general 😜

Reply:
Hmm I see. They are indeed "optimized". Not sure which of the alternative names is more discoverable/intuitive. I am more inclined to "constrained" but cant really say why xD

Reply:
I'll go with "constrained"

Reply:
Done